File: app/src/main/java/com/moggot/findmycarlocation/di/module/AppModule.java
package com.moggot.findmycarlocation.di.module;

import android.content.Context;

import com.moggot.findmycarlocation.App;

import javax.inject.Singleton;

import dagger.Module;
import dagger.Provides;

@Module
public class AppModule {

    @Provides
    @Singleton
    Context provideContext(App application) {
        return application.getApplicationContext();
    }
}

File: app/src/main/java/com/moggot/findmycarlocation/di/module/AppModule.kt
package com.moggot.findmycarlocation.di.module

import android.app.Application
import android.content.Context
import com.moggot.findmycarlocation.App
import dagger.Module
import dagger.Provides
import dagger.hilt.InstallIn
import dagger.hilt.components.SingletonComponent
import javax.inject.Singleton

@Module
@InstallIn(SingletonComponent::class)
class AppModule {

    @Singleton
    @Provides
    fun provideContext(application: Application): Context = application

    @Provides
    @Singleton
    fun provideApplication(application: Application): App {
        return application as App
    }
}
